### PR TITLE
Updates for 3-node deployment and destroy

### PR DESCRIPTION
**Pull request by ypise**
Signed-off-by: Yashodhan Pise <yashodhan.pise@seagate.com>

### CODE REVIEW

File: cli/src/cortx_prep.sh [hunk 1]
```diff
@@ -70,11 +70,20 @@ gpgcheck=0
 name=Repository Base
 enabled=1
 EOF
+
+        touch /etc/pip.conf
+cat <<EOF >/etc/pip.conf
+[global]
+timeout: 60
+index-url: $CORTX_RELEASE_REPO/python_deps/
+trusted-host: $CORTX_RELEASE_REPO
```
Comment (andkononykhin2) [inline cli/src/cortx_prep.sh:79]: does that make sense for local python index? In our states [we consider](https://github.com/Seagate/cortx-prvsnr/blob/197c8e4df5196c5c6ea88c5b47cc8813dd55c397/api/python/provisioner/srv/salt/repos/pip_config.sls#L27-L29) that only for remote ones

File: devops/ci/multi_node_deploy/Jenkinsfile_3Node_Deploy [hunk 2]
```diff
@@ -133,14 +133,25 @@ node {
...
                     yum-config-manager --add-repo "${CORTX_RELEASE_REPO}/3rd_party/"
-                    yum install --nogpgcheck -y python3 python36-m2crypto salt salt-master salt-minion
-                    rm -rf /etc/yum.repos.d/*3rd_party*.repo
                     yum-config-manager --add-repo "${CORTX_RELEASE_REPO}/cortx_iso/"
+                    
+                    cat <<EOF >/etc/pip.conf
+                    [global]
+                    timeout: 60
+                    index-url: $CORTX_RELEASE_REPO/python_deps/
+                    trusted-host: $(echo $CORTX_RELEASE_REPO | awk -F '/' '{print $3}')
```
Comment (andkononykhin2) [inline devops/ci/multi_node_deploy/Jenkinsfile_3Node_Deploy:142]: the same question here

File: devops/ci/multi_node_deploy/Jenkinsfile_3Node_Deploy [hunk 3]
```diff
@@ -155,7 +166,7 @@ node {
                 sshCommand remote: remotes[0], command: """
                     yum install -y sshpass
                     export SSHPASS=${PASSWORD}
```
Comment (andkononykhin2) [inline devops/ci/multi_node_deploy/Jenkinsfile_3Node_Deploy:168]: not related to the change but ... for such cases makes sense to consider [withCredentials](https://www.jenkins.io/doc/pipeline/steps/credentials-binding/)

File: devops/ci/single_node_deploy/Jenkinsfile_Deploy [hunk 4]
```diff
@@ -77,13 +77,24 @@ node {
...
                     yum-config-manager --add-repo "${CORTX_RELEASE_REPO}/3rd_party/"
-                    yum install --nogpgcheck -y python3 python36-m2crypto salt salt-master salt-minion
-                    rm -rf /etc/yum.repos.d/*3rd_party*.repo
                     yum-config-manager --add-repo "${CORTX_RELEASE_REPO}/cortx_iso/"
+                    
+                    cat <<EOF >/etc/pip.conf
+                    [global]
+                    timeout: 60
+                    index-url: $CORTX_RELEASE_REPO/python_deps/
+                    trusted-host: $(echo $CORTX_RELEASE_REPO | awk -F '/' '{print $3}')
```
Comment (andkononykhin2) [inline devops/ci/single_node_deploy/Jenkinsfile_Deploy:86]: the same question

**Review by andkononykhin2 — APPROVED** [review]
@ypise Looks good. Thank you (But I posted few questions.)

File: cli/src/cortx_prep.sh (hunk 1 above)
Comment (ypise) [reply cli/src/cortx_prep.sh:79]: Yes @andkononykhin2, it's a new requirement that follows the cortx-preq rpm from RE.

File: devops/ci/multi_node_deploy/Jenkinsfile_3Node_Deploy (hunk 2 above)
Comment (ypise) [reply devops/ci/multi_node_deploy/Jenkinsfile_3Node_Deploy:142]: Yes @andkononykhin2, it's a new requirement that follows the cortx-preq rpm from RE.

File: devops/ci/multi_node_deploy/Jenkinsfile_3Node_Deploy (hunk 3 above)
Comment (ypise) [reply devops/ci/multi_node_deploy/Jenkinsfile_3Node_Deploy:168]: We want the password to be passed as an argument from Build Parameters. WithCredentials might help us hiding the password. In fact, so can the https://plugins.jenkins.io/mask-passwords/ plugin.
Let me check if we can have that.

File: devops/ci/single_node_deploy/Jenkinsfile_Deploy (hunk 4 above)
Comment (ypise) [reply devops/ci/single_node_deploy/Jenkinsfile_Deploy:86]: Yes @andkononykhin2, it's a new requirement that follows the cortx-preq rpm from RE.

**Review by sradhanand-pati — APPROVED** [review]

File: devops/ci/multi_node_deploy/3node_deploy.jenkinsfile [hunk 5]
```diff
@@ -155,14 +166,14 @@ node {
                 sshCommand remote: remotes[0], command: """
                     yum install -y sshpass
                     export SSHPASS=${PASSWORD}
-                    sshpass -e provisioner setup_provisioner ${nodes} --logfile --logfile-filename /var/log/seagate/provisioner/setup.log --source rpm --config-path ~/config.ini --ha --dist-type bundle --target-build ${CORTX_RELEASE_REPO}
+                    provisioner setup_provisioner ${nodes} --logfile --logfile-filename /var/log/seagate/provisioner/setup.log --source rpm --config-path ~/config.ini --ha --dist-type bundle --target-build ${CORTX_RELEASE_REPO}
                     provisioner configure_setup /root/config.ini 1
```
Comment (sradhanand-pati) [inline devops/ci/multi_node_deploy/3node_deploy.jenkinsfile:170]: Can you update the '1' to '3' ?
for 3 node deployment

**Review by sradhanand-pati — CHANGES_REQUESTED** [review]

File: devops/ci/multi_node_deploy/3node_deploy.jenkinsfile [hunk 6]
```diff
@@ -79,11 +79,11 @@ node {
...
                     try { 
                         sshCommand remote: remote, command: """
-                            test 2 == \$(lsblk -d|grep -E 'sdb|sdc'|wc -l)
+                            test 6 -le $(lsblk -nd -o NAME -e 11|grep -v sda|wc -l)
                         """
-                        echo "The VM has exactly 2 nos. of attached disks. Check Successful!"
+                        echo "The VM has 6+ number of attached disks. Check Successful!"
                     } catch(Exception ex) {
-                        error 'The VM should have exactly 2 attached disks. Kindly provide a VM with exactly 2 attached disks.'
+                        error 'The VM should has less than 6 attached disks. Kindly provide a VM with 6+ attached disks.'
```
Comment (83bhp) [inline devops/ci/multi_node_deploy/3node_deploy.jenkinsfile:86]: "The VM should has less than 6" should be "The VM has less than 6..."

**Review by 83bhp — APPROVED** [review]
one minor comment, rest looks fine.

**Review by sradhanand-pati — APPROVED** [review]

File: devops/ci/multi_node_deploy/3node_deploy.jenkinsfile (hunk 5 above)
Comment (ypise) [reply devops/ci/multi_node_deploy/3node_deploy.jenkinsfile]: Addressed

**Pull request merged**